packages/back-end/src/services/vercel.ts

```diff
@@ -0,0 +1,143 @@
+import fetch from "node-fetch";
+import {
+  CreateEnvParams,
+  GbVercelEnvMap,
+  VercelEnvVar,
+  VercelProject,
+  VercelTarget,
+} from "../../types/vercel";
+import { createApiKey } from "./apiKey";
+
+const gbApiKey = {
+  key: "GROWTHBOOK_KEY",
+  description:
+    "This key allows you to connect your GrowthBook sdk to the GrowthBook API.",
+};
+const gbWebhookKey = {
+  key: "GROWTHBOOK_WEBHOOK_SECRET",
+  description: "This key allows you to connect to GrowthBook webhooks.",
+};
+const gbKeys = [gbApiKey, gbWebhookKey];
+
+interface VercelApiCallProps {
+  token: string;
+  teamId: string | null;
+  endpoint: string;
+  method: "POST" | "GET";
+  body?: string;
+}
+
+async function vercelApiCall<T = unknown>({
+  token,
+  teamId,
+  endpoint,
+  method,
+  body,
+}: VercelApiCallProps): Promise<T> {
+  if (teamId) {
+    endpoint = endpoint + `?${new URLSearchParams({ teamId })}`;
+  }
+  const res = await fetch(`https://api.vercel.com${endpoint}`, {
+    method,
+    body,
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${token}`,
+    },
+  });
+
+  const json: T | { error: string } = await res.json();
+  if ("error" in json) {
+    console.error(json.error);
+    throw new Error(json.error);
+  }
+  return json;
+}
+
+type ReducedMapping = { gb: string; vercel: VercelTarget[] };
+
+//Takes env map entries that have the same GB environment and makes them into a single entry by joining the Vercel environments
+export function reduceGbVercelEnvMap(gbVercelEnvMap: GbVercelEnvMap) {
+  const newEnvMap: Record<string, ReducedMapping> = {};
+
+  gbVercelEnvMap.forEach(({ gb, vercel }) => {
+    if (!gb) return;
+    newEnvMap[gb] = newEnvMap[gb] || { gb, vercel: [] };
+    newEnvMap[gb].vercel.push(vercel);
+  });
+
+  return Object.values(newEnvMap);
+}
+
+export async function createOrgGbKeys(
+  orgId: string,
+  gbVercelEnvMap: { gb: string; vercel: VercelTarget[] }[]
+) {
+  const orgGbKeys = [];
+  for (const envMapEntry of gbVercelEnvMap) {
+    for (const gbKey of gbKeys) {
```

Comment:
If we remove webhooks from the PR, `gbKeys` will just have a single element in it.  I would get rid of those constants and just hard-code the single key/description here:

```ts
for (const envMapEntry of gbVercelEnvMap) {
  const createdKeyVal = await createApiKey(
    orgId,
    envMapEntry.gb,
    "This key allows you to connect your GrowthBook sdk to the GrowthBook API."
  );
  orgGbKeys.push({
    key: "GROWTHBOOK_KEY",
    value: createdKeyVal,
    gbEnv: envMapEntry.gb,
    vercelEnvArr: envMapEntry.vercel,
  });
}
```